### PR TITLE
Add tutorial material

### PR DESCRIPTION
**Pull request by akaiDing**
introduction to some basic idea of zydeco.

### CODE REVIEW

File: docs/tutorial/intro_to_zydeco.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# Intro to Zydeco
+## Types
+Zydeco has two main types: value type (`A`) and computation type (`B`).
```
Comment (maxsnew) [inline docs/tutorial/intro_to_zydeco.md:3]: two *kinds* of types

Comment (maxsnew) [reply docs/tutorial/intro_to_zydeco.md:3]: Then I would also add something to think about what they mean: A value type classifies inert data and a computation type classifies programs that compute things

File: docs/tutorial/intro_to_zydeco.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+## Types
+Zydeco has two main types: value type (`A`) and computation type (`B`). 
+
+Value type can be: 
+- `X`: type variable defined using `data`.
+- `Int | String | Boolean | Unit`: basic primitive type with built-in functions.
+- `Thunk(B)`: suspend a computation type and consider it as a value type. 
+
+Computation type can be:
+- `X`: type variable defined using `codata`.
```
Comment (maxsnew) [inline docs/tutorial/intro_to_zydeco.md:11]: In line with the `A/B` convention I would use `X` for value type variables and `Y` for computation type variables.

File: docs/tutorial/intro_to_zydeco.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+- `Ret(A)`: one of computation types we can run, the other is `OS`.
```
Comment (maxsnew) [inline docs/tutorial/intro_to_zydeco.md:12]: `Ret(A)` classifies computations that return `A` values and `OS` classifies computations that can be run as a process that interacts with the `OS`.

**Review by ricky136973 — APPROVED** [review]
LGTM.

**Pull request merged**